frontend/packages/dev-console/src/components/user-preferences/SecureRouteFields.tsx

```diff
@@ -0,0 +1,190 @@
+import * as React from 'react';
+import {
+  Checkbox,
+  FormGroup,
+  Select,
+  SelectOption,
+  SelectVariant,
+  Skeleton,
+} from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import {
+  InsecureTrafficTypes,
+  PassthroughInsecureTrafficTypes,
+  TerminationTypes,
+} from '../import/import-types';
+import { usePreferredRoutingOptions } from './usePreferredRoutingOptions';
+
+import './SecureRouteFields.scss';
+
+const SecureRouteFields: React.FC = () => {
+  const { t } = useTranslation();
+  const [
+    preferredRoutingOptions,
+    setPreferredRoutingOptions,
+    preferredRoutingOptionsLoaded,
+  ] = usePreferredRoutingOptions();
+  const {
+    secure,
+    tlsTermination: prefferedTLSTermination,
+    insecureTraffic: prefferedInsecureTraffic,
+  } = preferredRoutingOptions;
+  const [secureRouteChecked, setSecureRouteChecked] = React.useState<boolean>(secure);
+  const [isTLSTerminationOpen, setIsTLSTerminationOpen] = React.useState<boolean>(false);
+  const [isInsecureTrafficOpen, setIsInsecureTrafficOpen] = React.useState<boolean>(false);
+  const [tlsTermination, setTlsTermination] = React.useState(prefferedTLSTermination);
+  const [insecureTraffic, setInsecureTraffic] = React.useState(prefferedInsecureTraffic);
```

Comment:
Same question here: Do we need this state or can we use the user preferences state instead?